### DISCUSSION TITLE
is there a way to hook into/customize Table layout

### DISCUSSION THREAD

**Discussion by Nomia** [directus/directus#18678] [Feature Requests]
### Summary

hi,

is there a way to hook into/customize Table layout, e.g. add an `operation` cell at the end of each data row

### Basic Example

add an `operation` cell at the end of each data row, like `delete`,`edit` etc.

### Motivation

If developers can hook into/customize the `table` layout, they don't have to rewrite a whole `layoyt` extension just to add a custom cell at the end of each row, etc.

### Detailed Design

since i have not read the source code of table layout, i don't know how to implement this 

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

no big performance impact, would enhance the extensibility of layout

### Alternatives

have to write the whole `layout` extension to implement some subtle change to the table layout

### Adoption Strategy

they will be happy to use this feature, or just ignore that

**Comment by rijkvanzanten** [comment]
Merging this with https://github.com/directus/directus/discussions/8941 as that's describing a way that could be used for a table-row slot as well 🙂
